car_cleaning_service_with_threads.rb

```diff
@@ -0,0 +1,115 @@
+require'time'
+
+class Car
+  attr_reader :id
+
+  def initialize(id)
+    @id = id
+  end
+
+  def to_s
+    @id
```

Comment:
Oh, I didn't realize that the @ is no longer required, but it makes sense! I overwrote the to_sym method to be able to use symbols as keys for the hash map containing Car objects, instead of the object itself. Is that a bad practice?

Reply:
Regarding overriding, it is not bad but I recommend to use it with caution in general. For example if you are overriding `to_s` method globally in a monkey patching then it might break other parts of code which relates to this method. So the comment was more like a note to be aware of this.

In your specific case `to_s` method returns just the id, without modifying it. So you could just use car.id instead of car.to_s

Reply:
I changed the implementation, thank you for the suggestion! It looks cleaner now.